pkg/scheduler/core/extender.go

```diff
@@ -299,8 +299,14 @@ func (h *HTTPExtender) Filter(
 
 	if h.nodeCacheCapable && result.NodeNames != nil {
 		nodeResult = make([]*v1.Node, 0, len(*result.NodeNames))
-		for i := range *result.NodeNames {
-			nodeResult = append(nodeResult, nodeNameToInfo[(*result.NodeNames)[i]].Node())
+		for _, nodeName := range *result.NodeNames {
+			if node, ok := nodeNameToInfo[nodeName]; ok {
+				nodeResult = append(nodeResult, node.Node())
+			} else {
+				result.FailedNodes[nodeName] = fmt.Sprintf(
```

Comment:
From the issue #79640, it seems that it is because the scheduler extender returns a node wrongly and the node is not passed by scheduler. It might be enough to add a log for it?

Reply:
I also try to just add log for it, but by convension, seems no log is used in extender.go. 
So, I turn to return it as a failed node, which can also be easily exposed to users, that why extender returned candidates is skipped.

Reply:
I also saw other simliar practice in this file, such as:
https://github.com/kubernetes/kubernetes/blob/b893a64b0e68d11db5959d5dd03af774ad3f02a3/pkg/scheduler/core/extender.go#L203

But it is for preemption, that needs all victims on a selected node to be preempted, so just return error is an easy way.
For filter, we can improve it by still picking other vaild candidates, because any one would be feasiable.

Reply:
Hi @hex108 any suggestions? :)

Reply:
`FailedNodes` will be displayed to users, and it is a little confused to display a node that does not exist. I think we need print the error message and fix it if there's cache inconsistency between default scheduler and extender.

For the `preemption`, it returns a error, and just log it finally at https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/scheduler.go#L307.

Reply:
Thanks @hex108 
I updated the PR, please check again.
BTW, I found the preempt error is not logged acutally, so also fixed it.